lib/trino-hive-formats/src/main/java/io/trino/hive/formats/ion/IonEncoderFactory.java

```diff
@@ -144,7 +144,10 @@ private void encodeStruct(IonWriter writer, IntFunction<Block> blockSelector, in
         {
             writer.stepIn(IonType.STRUCT);
             for (int i = 0; i < fieldEncoders.size(); i++) {
-                // todo: the Hive SerDe omits fields when null by default
+                // Omit the filed when the field is null
+                if (blockSelector.apply(i).isNull(position)) {
```

Comment:
super nit: I would create a local variable to capture the `BlockBuilder` instead of calling `apply` on the `blockSelector` twice. Simpler and faster.